### ISSUE TITLE
inplace processing

### ISSUE THREAD

**Issue by alexanderlerch** [SileYin/2022-MUSI6106#11]
https://github.com/SileYin/2022-MUSI6106/blob/ea1599076db7172ca9f36924052d748d09c63c26/src/CombFilter/CombFilter.cpp#L49

note that if you change the get and put order, this also works inplace, i.e. for ppfInputBuffer == ppfOutputBuffer

**Comment by SileYin** [comment]
Yes. It's reasonable to set the buffer length to maxDelayLength+1 and change the order or put and get to have inplace processing rather than saving one byte for a tricky max length processing
